### DISCUSSION TITLE
How to remove dashes from SSN

### DISCUSSION THREAD

**Discussion by EthanKC** [nextgenhealthcare/connect#5132] [Q&A] [answered]
I'm trying to look for the best way to remove dashes from the SSN in Mirth Connect. I assume it's something in javascript, but I'm still figuring that out. Any ideas?

**Comment by pacmano1** [comment, marked as answer]
That is a simple replace in javascript or java, you should see many examples over at the forums.  StackOverflow has examples also.

What did you try that is not working?

**Reply by EthanKC** [reply]
I was still working that out. Was trying to do a substring setup, but of course replace is the way to go. Does this look right? 

`var SSN=msg['PID']['PID.19'].toString()

if(SSN.length > 0)
	if(SSN.includes("-")) = "true"
		msg['PID']['PID.19'] = SSN.replace(SSN,"")`

**Reply by pacmano1** [reply]
`$c('SSN',msg['PID']['PID.19']['PID.19.1'].toString().replace(/-/g,""))`

After sending a message through look at the "Mappings" tab in the message viewer and look for the channel map variable named SSN.

**Reply by pacmano1** [reply]
![image](https://user-images.githubusercontent.com/44065187/162830291-21b5793b-574e-4d3f-8d1e-d3c2f147d003.png)

Using a mapper step to do the same thing.

**Reply by EthanKC** [reply]
Ah ok, that use of the mapper makes sense. I honestly forgot about the string replacement aspect of that type. 

For the javascript, to make sure I'm understanding: 
$c - that tells it to make it a channel map variable, right? 
/-/g - this is a regular expression that makes sure it replaces all instances of "-", not just the first? 

If I used the javascript, would I also need to have a second step that sets the field equal to the "SSN" variable? I'm going to go with the mapper step, since it's simpler, but the more I can understand the javascript the better.

**Reply by pacmano1** [reply]
Yes on your first two questions.  I would expect over time you would stop using mapper steps because you just end up scrolling through a bunch of them.   You do need to be mindful of a javascript variable vs a java variable - best thing to do is just mess with them and see what works and doesn't work.
